### PR TITLE
Add Animated.template

### LINKED ISSUE #29: Animate values inside a calc() expression
Is it possible to animate values inside calc expressions? For example, something like this:

```
width: calc(300px + ${progress} * (100% - 300px));
```

### PR DESCRIPTION
**Pull request by arcanis**
This PR should fix #29, and provide a workaround for #31.

I've had some issues with running the tests (jest was yelling at me about missing modules), but I commented everything else except my new tests, fixed the few remaing require calls, and it eventually passed as expected. I'd still be interested to know how to run the tests properly.

### CODE REVIEW

File: src/AnimatedTemplate.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+ * @flow
+ */
+'use strict';
+
+var Animated = require('./Animated');
+var AnimatedWithChildren = require('./AnimatedWithChildren');
+
+class AnimatedTemplate extends AnimatedWithChildren {
+  _strings: Array<string>;
+  _valies: Array;
```
Comment (vjeux) [inline src/AnimatedTemplate.js]: Values

File: src/AnimatedTemplate.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    if (value instanceof Animated) {
+      return value.__getValue();
+    } else {
+      return value;
+    }
+  }
+
+  __getValue(): String {
+    var value = this._strings[0];
+    for (var t = 0, T = this._values.length; t < T; ++t) {
```
Comment (vjeux) [inline src/AnimatedTemplate.js]: nit: could you write it

``` js
for (var i = 0; i < this._values.length; ++i) {
```

in order to be consistent with the rest of the project

File: src/index.js [hunk 3]
```diff
@@ -442,6 +443,14 @@ module.exports = {
   },
 
   /**
+   * Creates a new Animated value that is the specified string, with each
+   * substitution expression being separately animated and cominterpolated.
+   */
+  template: function template(strings, ... values) {
```
Comment (vjeux) [inline src/index.js]: nit: no space after `...`

File: src/index.js [hunk 4]
```diff
@@ -442,6 +443,14 @@ module.exports = {
   },
 
   /**
+   * Creates a new Animated value that is the specified string, with each
+   * substitution expression being separately animated and cominterpolated.
```
Comment (vjeux) [inline src/index.js]: "cominterpolated" is this a real word? :p

File: src/__tests__/Animated-test.js [hunk 5]
```diff
@@ -401,6 +401,25 @@ describe('Animated Interactions', () => {
...
+    var template = Animated.template `width: 100px`;
+    expect(template.__getValue()).toBe('width: 100px');
+  });
+  it('works with literal template that uses static values', () => {
+    var value = 42;
+    var template = Animated.template `width: ${value}px; height: ${value}px`;
+    expect(template.__getValue()).toBe('width: 42px; height: 42px');
+  });
+  it('works with literal template that uses Animated values', () => {
+    var value = new Animated.Value(42);
```
Comment (vjeux) [inline src/__tests__/Animated-test.js:411]: could you add a test with two distinct animated values?

**Comment by vjeux** [comment]
This is awesome, thanks for doing this! Once you add the small test i'm requesting, i'll merge it in :)

Also, could you sign the fb cla: https://code.facebook.com/cla thanks!

File: src/index.js (hunk 4 above)
Comment (arcanis) [inline src/index.js]: Hm, I fear the world is not ready for it yet :smile:

**Comment by arcanis** [comment]
Happy to contribute! I've added the additional requested test :)

**Comment by vjeux** [comment]
Awesome, thanks a lot :) cc @lelandrichardson to do a new release and figure out why the tests are broken.

**Pull request merged**